Add unit tests to device tree crate.

doriath opened the pull request:
Signed-off-by: Tomasz Zurkowski <zurkowski@google.com>

rminnich commented on src/lib/device_tree/src/lib.rs:
```diff
@@ -38,27 +69,41 @@ enum Token {
 }
 
 impl Path {
+    /// Returns the number number of components in the path.
```
number number -> number

doriath replied:
I am very sorry, somehow changes in this PR got some unrelated changes in this file, reverted all of them to prevent the conflict with the other PR.

rminnich commented on src/lib/device_tree/src/lib.rs:
```diff
@@ -92,10 +137,10 @@ fn align4(x: usize) -> usize {
 impl<'a> FdtReader<'a> {
     pub fn new(drv: &'a dyn Driver) -> Result<FdtReader<'a>> {
+        // TODO: Instead of reading each number one by one consider
+        // reading the whole size of the header at once, and then just
```
and then just ... ?

doriath replied:
I am very sorry, somehow changes in this PR got some unrelated changes in this file, reverted all of them to prevent the conflict with the other PR.

rminnich commented:
where are we on this. Ryan? It seems useful.

doriath commented:
I believe I addressed all the comments and I was waiting for next round of review (if I missed something that I should do / fix / improve please let me know).

rjoleary reviewed (approved):
This looks awesome! Thank you
